sklearn/linear_model/_coordinate_descent.py

```diff
@@ -703,18 +707,44 @@ def fit(self, X, y, check_input=True):
             y = check_array(y, order='F', copy=False, dtype=X.dtype.type,
                             ensure_2d=False)
 
+        n_samples, n_features = X.shape
+        alpha = self.alpha
+
+        if isinstance(sample_weight, numbers.Number):
+            sample_weight = None
+        if sample_weight is not None:
+            if check_input:
+                if sparse.issparse(X):
+                    raise ValueError("Sample weights do not (yet) support "
+                                     "sparse matrices.")
+                sample_weight = _check_sample_weight(sample_weight, X,
+                                                     dtype=X.dtype)
+            # simplify things by rescaling sw to sum up to n_samples
+            # => np.average(x, weights=sw) = np.mean(sw * x)
+            sample_weight *= (n_samples / np.sum(sample_weight))
+            # Objective function is:
+            # 1/2 * np.average(squared error, weights=sw) + alpha * penalty
+            # but coordinate descent minimizes:
+            # 1/2 * sum(squared error) + alpha * penalty
+            # enet_path therefore sets alpha = n_samples * alpha
+            # With sw, enet_path should set alpha = sum(sw) * alpha
+            # Therefore, we rescale alpha = sum(sw) / n_samples * alpha
+            # Note: As we rescaled sample_weights to sum up to n_samples,
+            #       we don't need this
+            # alpha *= np.sum(sample_weight) / n_samples
+
         # Ensure copying happens only once, don't do it again if done above
         should_copy = self.copy_X and not X_copied
         X, y, X_offset, y_offset, X_scale, precompute, Xy = \
             _pre_fit(X, y, None, self.precompute, self.normalize,
                      self.fit_intercept, copy=should_copy,
-                     check_input=check_input)
+                     check_input=check_input, sample_weight=sample_weight,
+                     order='F')
```

Comment:
Maybe worth adding a comment about why `order="F"` is more efficient here

Reply:
It's only more efficient code at this place, not better performance.